lib/reversetunnel/agentpool.go

```diff
@@ -50,13 +50,19 @@ type AgentPool struct {
 	log          *log.Entry
 	cfg          AgentPoolConfig
 	proxyTracker *track.Tracker
-	ctx          context.Context
-	cancel       context.CancelFunc
+
+	// ctx controls the lifespan o the agent pool, and is used to control
+	// all of the sub-processes it spawns.
+	ctx    context.Context
+	cancel context.CancelFunc
 	// spawnLimiter limits agent spawn rate
 	spawnLimiter utils.Retry
 
 	mu     sync.Mutex
 	agents map[utils.NetAddr][]*Agent
+
+	// waitResult is a value that will be returned to the caller of Wait
+	result error
```

Comment:
nit: Rename either comment or variable? (Renaming the variable seems better)

```suggestion
	// waitResult is a value that will be returned to the caller of Wait
	waitResult error
```